library/src-bootstrapped/scala/internal/quoted/Expr.scala

```diff
@@ -20,7 +20,7 @@ import scala.internal.tasty.CompilerInterface.quoteContextWithCompilerInterface
   }
 
   def unseal(using qctx: QuoteContext): qctx.tasty.Term =
-    if (quoteContextWithCompilerInterface(qctx).tasty.compilerId != scopeId)
+    if (qctx.hashCode != scopeId)
```

Comment:
Is it possible to do the check lazily inside the compiler, so that we can avoid changing the semantics of `hashCode`?

Reply:
Indeed, we should lazily compute this hash. But we should change the `hashCode` as this is semantically a better fit for it.

Reply:
Either way the idea is to eventually remove it. If we decide to keep it we can figure out a better efficient scheme at that point.

Reply:
It seems we are just repurposing the method `hashCode` to do something else in this PR, to avoid creating another method. This raises concern.

I'm wondering if we can remove the checks in the library, and only perform checks in the compiler, so that we may not need the repurposing.

Reply:
Before we were using another method to return the hash associated with that QuoteContext. Now we use hashCode to return the hash of that instance.

Reply:
And #8940 aims to remove this altogether.